qiskit/algorithms/gradients/base_qfi.py

```diff
@@ -0,0 +1,182 @@
+        if parameters is None:
+            parameters = [None] * len(circuits)
+        # Validate the arguments.
+        self._validate_arguments(circuits, parameter_values, parameters)
+        # The priority of run option is as follows:
+        # options in ``run`` method > QFI's default options > primitive's default setting.
+        opts = copy(self._default_options)
+        opts.update_options(**options)
+        job = AlgorithmJob(self._run, circuits, parameter_values, parameters, **opts.__dict__)
+        job.submit()
+        return job
+
+    @abstractmethod
+    def _run(
+        self,
+        circuits: Sequence[QuantumCircuit],
+        parameter_values: Sequence[Sequence[complex]],
+        parameters: Sequence[Sequence[Parameter] | None],
+        **options,
+    ) -> QFIResult:
+        """Compute the QFIs on the given circuits."""
+        raise NotImplementedError()
+
+    def _validate_arguments(
+        self,
+        circuits: Sequence[QuantumCircuit],
+        parameter_values: Sequence[Sequence[complex]],
+        parameters: Sequence[Sequence[Parameter] | None] | None = None,
+    ) -> None:
+        """Validate the arguments of the ``run`` method.
+
+        Args:
+            circuits: The list of quantum circuits to compute the QFIs.
+            parameter_values: The list of parameter values to be bound to the circuit.
+            parameters: The Sequence of Sequence of Parameters to calculate only the QFIs of
+                the specified parameters. Each Sequence of Parameters corresponds to a circuit in
+                ``circuits``. Defaults to None, which means that the QFIs of all parameters in
+                each circuit are calculated.
+
+        Raises:
+            ValueError: Invalid arguments are given.
+        """
+        # Validation
+        if len(circuits) != len(parameter_values):
+            raise ValueError(
+                f"The number of circuits ({len(circuits)}) does not match "
+                f"the number of parameter value sets ({len(parameter_values)})."
+            )
+
+        if parameters is not None:
+            if len(circuits) != len(parameters):
+                raise ValueError(
+                    f"The number of circuits ({len(circuits)}) does not match "
+                    f"the number of the specified parameter sets ({len(parameters)})."
+                )
+
+        for i, (circuit, parameter_value) in enumerate(zip(circuits, parameter_values)):
+            if not circuit.num_parameters:
+                raise ValueError(f"The {i}-th circuit is not parameterised.")
+            if len(parameter_value) != circuit.num_parameters:
+                raise ValueError(
+                    f"The number of values ({len(parameter_value)}) does not match "
+                    f"the number of parameters ({circuit.num_parameters}) for the {i}-th circuit."
+                )
+
+    @property
+    def options(self) -> Options:
```

Comment:
Since this is just forwarding to another method, would it be simpler to just have the `options` property be abstract?